### ISSUE TITLE
[RFC] Abandon FrameworkExtraBundle

### ISSUE THREAD

**Issue by derrabus** [symfony/symfony#44705]
### Description

Since Symfony 2 `sensio/framework-extra-bundle` is the solution for adding annotation-magic to the application. Since then, some functionality has been moved to the core (e.g. annotation-based routing) and other functionality has been obsoleted by a core feature (param converters Vs. argument value providers).

FrameworkExtraBundle feels like a core-bundle but it's not as well maintained as we would expect a core bundle to be.

I think, it would be a good idea to review the feature set of the bundle and move commonly used functionality to the monorepo. Speaking for myself, that would be the security annotations like `IsGranted` as well as the Doctrine entity param converter (see #43854).

What do you think?

**Comment by fabpot** [comment]
I agree.

**Comment by garak** [comment]
I don't get the logic. There's a set of optional features in a bundle that is not well maintained. The solution should be maintaining that bundle, not putting optional features in the core.

I think a strong point of Symfony is having a slim base and modularity, I hope it's not going to transform in a copy of Laravel.

**Comment by wouterj** [comment]
Let's not act as if moving 3 annotations as native PHP attributes to core will upset anyone please.

That being said, "the solution should be maintaining the bundle" is nicely said but not the practice. Apparently nobody is interest in maintaining the bundle (for years now).
So either it will be dead, we'll remove them from the official docs and that's it. Or we see value in some features and move the few attributes to core.

I'm all +1 for the latter.

**Comment by javiereguiluz** [comment]
@garak generally speaking I agree with you ... but in this particular case, I think the bundle was "wrong" since day one. I mean, these were so useful and common features that I never felt them as "extra features". In my opinion, they should've been "core/built-in features" since day one.

**Comment by fabpot** [comment]
These feature were in a bundle as annotations in phpdocs were controversial back  then. Now that we have native annotations, it makes sense to move them to core. Add simple as that.

**Comment by Nyholm** [comment]
I agree with this as well.  

@derrabus will you do/coordinate the work?

**Comment by stof** [comment]
As several features have not been migrated yet to the core, we cannot abandon the bundle yet (we probably cannot abandon it fully before the end of the 5.4 LTS, as the 5.4 branch won't have replacements).

However, I'm also in favor of bringing the remaining features to the core to abandon the bundle in the future

**Comment by carsonbot** [comment]
Thank you for this suggestion.
There has not been a lot of activity here for a while. Would you still like to see this feature?

**Comment by stof** [comment]
@carsonbot keep this opened. The work is in progress in various places

**Comment by nicolas-grekas** [comment]
Help wanted (see #45415):

- [x] Cache #46880 
- [x] Template #46906
- [x] IsGranted #46907 (see #46912)
- [x] ~Security~ (see #46912)
- [x] Entity #43854
- [x] ~ParamConverter~ (see below)

**Comment by nicolas-grekas** [comment]
>  ParamConverter :arrow_right: Symfony\Component\HttpKernel\Attribute\ParamConverter

We don't need this one, do we?

**Comment by stof** [comment]
indeed, I don't think we need it

**Comment by chalasr** [comment]
The main features have been migrated to core. Let's iterate and deprecate FrameworkExtraBundle :)

**Comment by stof** [comment]
Well, let's wait for the release of the core feature first before deprecating FrameworkExtraBundle

**Comment by arderyp** [comment]
@stof I know this is an old closed issue but I have a relevant question.

I am on the latest 5.4.x. I just migrated from annotations to attributes.  Everything is working great. grep is returning no instances of `Sensio` or `sensio` anywhere in my code. I dropped by sensio_framework_extra_bundle config. Everything is working.

I do not use any of the attributes introduced in 6.2 such as `MapEntity`, so 5.4.x core attributes seem to handle my application.

And yet... despite all of the above, when I remove the SensioFrameworkExtraBundle entry from `config/bundles.php`, my app breaks completely with errors about being unable to map entity arguments in controller route parameters ([like this](https://stackoverflow.com/questions/66657307/cannot-autowire-argument-article-symfony-5)). My `config/services.php` loads `src/` but excludes the `Kernel.php` and all associated Entity directories. As far as I know, this is the advised approach per documentation. If I un-exclude the Entity directories, this error is fixed, but then a whole new issue arises with object persistence, which would not be entirely unexpected if the entities are registered as services.

Am I missing something, or is it literally not possible to drop the SensioFrameworkExtraBundle package/dependency until I migrate up to 6.2 (which I will be working on in the coming weeks)?

Thanks in advance.

**Comment by arderyp** [comment]
@stof I know this is an old closed issue but I have a relevant question.

I am on the latest 5.4.x. I just migrated from annotations to attributes.  Everything is working great. grep is returning no instances of `Sensio` or `sensio` anywhere in my code. I dropped by sensio_framework_extra_bundle config. Everything is working.

And yet... when I remove the SensioFrameworkExtraBundle entry from `config/bundles.php`, my app breaks many routes with errors about being unable to map entity arguments in controller route parameters ([like this](https://stackoverflow.com/questions/66657307/cannot-autowire-argument-article-symfony-5)). 

It looks like `ParamConverterListener` is the source of my problem.  Does Symfony not have a native replacement for that until 6.2?
